File: leetcode_problems/cpp/lc1491_avg_salary_exc_min_and_max.cpp
#include <vector>
#include <iostream>
#include <limits>
using std::vector;

class Solution
{
public:
    double average(vector<int> &salary)
    {
        int maxi = std::numeric_limits<int>::min();
        int mini = std::numeric_limits<int>::max();
        int sum = 0;
        for (int i = 0; i < salary.size(); i++)
        {

            int cur = salary[i];
            if (cur > maxi)
            {
                maxi = cur;
            }
            if (cur < mini)
            {
                mini = cur;
            }
            sum += cur;
        }
        sum -= mini;
        sum -= maxi;
        return static_cast<double>(sum) / (salary.size() - 2);
    }
};

int main()
{
    // vector<int> salary = {4000, 3000, 1000, 2000};
    vector<int> salary = {1000, 2000, 3000};
    Solution sol;
    double result = sol.average(salary);
    std::cout << result << std::endl;
    return 0;
}

File: leetcode_problems/typescript/src/lc1491_avg_salary_exc_min_and_max.ts
function average(salary: number[]): number {
  let maxi = Number.MIN_SAFE_INTEGER;
  let mini = Number.MAX_SAFE_INTEGER;
  let sum = 0;
  for (let num of salary) {
    if (num > maxi) {
      maxi = num;
    }
    if (num < mini) {
      mini = num;
    }
    sum += num;
  }
  return (sum - mini - maxi) / (salary.length - 2);
}

let salary = [4000, 3000, 1000, 2000];
console.log(average(salary));
